terraform/cross-device/workload_identity.tf

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module "gke-workload-identity" {
```

Comment:
Don't use the workload-identity module because you don't need to create any service accounts, and you don't need to configure Workload Identity because [it's already configured](https://github.com/GoogleCloudPlatform/federated-learning/blob/d8fed2b356a32eb4799977416e06623d6e410ca2/terraform/iam.tf#L63).

I think you need to grant these roles to the service account that you're going to pass using the variable I proposed (see my other comment). To do this, use the `projects_iam` module. [Example](https://github.com/GoogleCloudPlatform/federated-learning/blob/d8fed2b356a32eb4799977416e06623d6e410ca2/terraform/iam.tf#L29)

Reply:
(once you do this, you probably want to rename this file to something other than `workload-identity.tf` because it's not going to be about Workload Identity. Maybe `iam.tf`?